pkg/controller/jobframework/base_webhook.go

```diff
@@ -55,7 +59,11 @@ func (w *BaseWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	job := w.FromObject(obj)
 	log := ctrl.LoggerFrom(ctx)
 	log.V(5).Info("Applying defaults", "job", klog.KObj(job.Object()))
-	ApplyDefaultForSuspend(job, w.ManageJobsWithoutQueueName)
+	if features.Enabled(features.KueueDefaulting) && w.Queues.DefaultLocalQueue(job.Object().GetNamespace()) != "" {
```

Comment:
Could we move this logic under `ApplyCommonDefaulting`. This will help to avoid the duplication